src/app/components/form/form.component.ts

```diff
@@ -233,9 +233,22 @@ export class FormComponent implements OnInit, OnChanges, OnDestroy {
     }
   }
 
+  // Remove trailing spaces and dots, and leading spaces
+  private sanitizeDomain(domain: string): string {
+    domain = domain.trim();
+    if (domain == '.') {
+      return domain;
+    } else {
+      return domain.replace(/[ \.]+$/, '');
```

Comment:
I'm slightly surprised by this regex. It seems it would normalize `zonemaster.net... . ..  .` to `zonemaster.net`. Is this intentional? I would have expected to see something like `/\.? +$/`, but maybe that's overly strict?

Reply:
Yes it was intentional but I realise now that this may not be the best behaviour. I have updated it.

Also now I the domains are not updated in the form after being sanitized, this way inputting `zonemaster.net. . . `  then clicking submit twice will always result in the same `zonemaster.net. . ` domain and not `zonemaster.net. . ` then `zonemaster.net. `.